src/redis_lock/__init__.py

```diff
@@ -29,15 +30,49 @@ class NotAcquired(RuntimeError):
     pass
 
 
+class AlreadyStarted(RuntimeError):
+    pass
+
+
 class Lock(object):
-    def __init__(self, redis_client, name, expire=None, id=None):
+    """
+    A Lock context manager implemented via redis SETNX/BLPOP.
+    """
+
+    def __init__(self, redis_client, name, expire=None, id=None, auto_renewal=False):
+        """
+        :param redis_client:
+            An instance of :class:`~StrictRedis`.
+        :param name:
+            The name (redis key) the lock should have.
+        :param expire:
+            The lock expiry time in seconds. If left at the default (None)
+            the lock will not expire.
+        :param id:
+            The ID (redis value) the lock should have. A random value is
+            generated when left at the default.
+        :param auto_renewal:
+            If set to True, Lock will automatically renew the lock so that it
+            doesn't expire for as long as the lock is held (acquire() called
+            or running in a context manager).
+
+            Implementation note: Renewal will happen using a daemon thread with
+            an interval of expire*2/3. If wishing to use a different renewal
+            time, subclass Lock, call super().__init__() then set
+            self._lock_renewal_interval to your desired interval.
+        """
         assert isinstance(redis_client, StrictRedis)
+        if auto_renewal and expire is None:
+            raise ValueError("Expire may not be None when auto_renewal is set")
```

Comment:
Maybe `TypeError` is more suitable than `ValueError` here?

TypeError =  Inappropriate argument type.
ValueError = Inappropriate argument value (of correct type).